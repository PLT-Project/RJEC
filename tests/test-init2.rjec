struct my_struct {
    a int;
    b char;
    c bool;
}

func foo(x int, y char) struct my_struct {
    i := struct my_struct {
        a : x,
        b : y
    };
    printi(i.a);
    printc(i.b);
    printb(i.c);

    return i;
}

func main() {

    t, k, l := foo(1, 'r'), foo(2, 'j'), 3;
    printi(t.a);
    printc(t.b);
    printb(t.c);

    printi(k.a);
    printc(k.b);
    printb(k.c);

    printi(l);

    a, b, c := t.a, t.b, t.c;
    printi(a);
    printc(b);
    printb(c);

}