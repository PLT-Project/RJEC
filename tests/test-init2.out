1
r
false
2
j
false
1
r
false
2
j
false
3
1
r
false